src/Nancy.Authentication.Stateless/StatelessAuthentication.cs

```diff
@@ -74,16 +74,10 @@ public static void Enable(INancyModule module, StatelessAuthenticationConfigurat
 
             return context =>
             {
-                try
-                {
-                    context.CurrentUser = configuration.GetUserIdentity(context);
-                    return context.Response;
-                }
-                catch (Exception)
-                {
-                    return context.Response;
-                }
+                context.CurrentUser = configuration.GetUserIdentity(context);
+
+                return null;

```

Comment:
You can see [here](https://github.com/NancyFx/Nancy/blob/665a59a23ce3df74672f3334bf653947151f11a3/src/Nancy/NancyEngine.cs#L239-L240), that if the pre-request hooks return a response, the request dispatcher (the whole module pipeline) will never be called. Further, you can see in the [BeforePipeline](https://github.com/NancyFx/Nancy/blob/665a59a23ce3df74672f3334bf653947151f11a3/src/Nancy/BeforePipeline.cs#L56-L60) that it short-circuits if the response is non-null.